Challenge-4/Bot_1/src/agent.spec.ts

```diff
@@ -0,0 +1,328 @@
+
+  const setupMockProvider = (collateralAddresses: string[]) => {
+    mockProvider.setNetwork(1);
+    collateralAddresses.forEach((address, index) => {
+      mockProvider.setNetwork(1);
+      mockProvider.addCallTo(mockAssetTokenAddress, 0, Iface, "getAssetInfo", {
+        inputs: [index],
+        outputs: [
+          index,
+          address,
+          createAddress("0x234"),
+          4,
+          12,
+          134,
+          1245,
+          1265,
+        ],
+      });
+      mockProvider.addCallTo(address, 0, getCollateralName, "name", {
+        inputs: [],
+        outputs: [`Collateral-${index + 1}`],
+      });
+    });
+  };
+
+  beforeEach(async () => {
+    mockProvider = new MockEthersProvider() as any;
+    mockProvider.setNetwork(1);
+    initialize = provideInitialize(
+      mockProvider as unknown as ethers.providers.Provider,
+    );
+
+    handleTransaction = provideHandleTransaction(
+      mockProvider as unknown as ethers.providers.Provider,
+      assetInfo,
+    );
+
+    helper = new Helper(mockProvider as unknown as ethers.providers.Provider);
+
+    txEvent = new TestTransactionEvent().setBlock(0);
+
+    await initialize();
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  it("should return single finding for single event occurence", async () => {
+    const collateralAddresses = [
+      createAddress("0x1247"),
+      createAddress("0x12567"),
+      createAddress("0x1289"),
+      createAddress("0x1290"),
+    ];
+
+    setupMockProvider(collateralAddresses);
+
+    txEvent.addEventLog(
+      "event SetGovernor(address indexed cometProxy, address indexed oldGovernor, address indexed newGovernor)",
+      mockConfigurator,
+      [...mockArgs],
+    );
+
+    const comparedNewAndOldValues: { [key: string]: any } = {
+      SetGovernor: {
+        Old_value: createAddress("0x123"),
+        New_value: createAddress("0x234"),
+      },
+    };
+
+    const mockAssetData = {
+      "Collateral Asset - Collateral-1": `${collateralAddresses[0]}`,
+      "Collateral Asset - Collateral-2": `${collateralAddresses[1]}`,
+      "Collateral Asset - Collateral-3": `${collateralAddresses[2]}`,
+      "Collateral Asset - Collateral-4": `${collateralAddresses[3]}`,
+    };
+
+    const findings = await handleTransaction(txEvent);
+
+    expect(findings.length).toStrictEqual(1);
+
+    expect(findings).toEqual([
+      Finding.fromObject({
+        name: `Change of asset values due to governance proposal`,
+        description: `The asset ${mockAssetTokenAddress} has been modified by a governance proposal`,
+        alertId: "GOV-1",
+        severity: FindingSeverity.Info,
+        type: FindingType.Info,
+        protocol: "Compound",
+        metadata: {
+          ...comparedNewAndOldValues,
+          ...mockAssetData,
+        },
+      }),
+    ]);
+  });
+
+  it("should return 0 findings for event other than proposal change", async () => {
+    mockProvider.setNetwork(1);
```

Comment:
Should this be done in the `initialize` or `beforeEach()` to not have to do several time?